pkg/apis/core/validation/validation.go

```diff
@@ -2228,6 +2237,17 @@ func ValidatePersistentVolumeClaimSpec(spec *core.PersistentVolumeClaimSpec, fld
 				"must match dataSourceRef"))
 		}
 	}
+	if spec.VolumeAttributesClassName != nil {
```

Comment:
Nit: can this be **if spec.VolumeAttributesClassName != nil && opts.EnableVolumeAttributesClass**?